scripts/null_route_helper

```diff
@@ -0,0 +1,216 @@
+
+    if not target_table:
+        error("Table {} not found, exiting...".format(table_name))
+
+    return target_table
+
+
+def get_acl_rule_key(ip_prefix):
+    """
+    Get the key that will be used to refer to the ACL rule used to block traffic from a source ip.
+    Since the rules are all given the same priority in SONiC, we can't identify a rule based on the priority.
+    So, we use the destination IP being blocked to give each rule a unique name in the system.
+    """
+    return ACL_RULE_PREFIX + str(ip_prefix)
+
+
+def get_all_acl_rules(configdb, table_name):
+    """
+    Return a dict of existed acl rules
+    {(u'SVI_PROTECT_V4', u'BLOCK_RULE_0.0.0.0/0'): {'PRIORITY': '9999', 'PACKET_ACTION': 'FORWARD', 'SRC_IP': '0.0.0.0/0', 'DST_IP': '0.0.0.0/0'},...}
+    """
+    key = CONFIG_DB_ACL_RULE_TABLE + '|' + table_name
+    all_rules = configdb.get_table(key)
+    block_rules = {}
+    for k, v in all_rules.items():
+        if k[1].startswith(ACL_RULE_PREFIX):
+            block_rules[k] = v
+
+    return block_rules
+
+
+def validate_input(ip_address):
+    """
+    Validate the format of input
+    """
+    try:
+        ver = ip_ver(ip_address)
+        if len(ip_address.split('/')) == 1:
+            ip_address += '/32' if ver == 4 else '/128'
+        return ip_address
+    except ValueError as e:
+        error("Could not parse {} as a valid IP address; exception={}".format(ip_address, e))
+
+
+def build_acl_rule(priority, src_ip):
+    """
+    Bild DROP rule for given src_ip and priority
+    """
+    rule = {
+        "PRIORITY": str(priority),
+        "PACKET_ACTION": "DROP"
+    }
+    if ip_ver(src_ip) == 4:
+        rule['ETHER_TYPE'] = str(0x0800)
+        rule['SRC_IP'] = src_ip
+    else:
+        rule['IP_TYPE'] = 'IPV6ANY'
+        rule['SRC_IPV6'] = src_ip
+
+    return rule
+
+
+def get_rule(configdb, table_name, ip_prefix):
+    """
+    Get Acl rule for given ip_prefix
+    """
+    key_name = 'SRC_IP' if ip_ver(ip_prefix) == 4 else 'SRC_IPV6'
+    all_rules = get_all_acl_rules(configdb, table_name)
+    for key, rule in all_rules.items():
+        if ip_prefix == rule[key_name]:
+            return {key: rule}
+
+    return None
+
+
+def update_acl_table(configdb, acl_table_name, ip_prefix, action):
+    """
+    Update ACL table to apply new rules for given ip_prefix. 'action' is supposed to be in ['DENY', 'ALLOW']
+    For 'DENY', an 'DROP' rule for given ip_prefix will be added if not existed
+    For 'ALLOW', we will try to remove the existing 'DENY' rule, and nothing is changed if not existed
+    """
+    check_table_existence(configdb, acl_table_name)
+    rule = get_rule(configdb, acl_table_name, ip_prefix)
+    rule_key = list(rule.keys())[0] if rule else None
+    rule_value = list(rule.values())[0] if rule else None
+    if action == ACTION_ALLOW:
+        if not rule:
+            return
+        configdb.mod_entry(CONFIG_DB_ACL_RULE_TABLE, rule_key, None)
```

Comment:
>mod_entry [](http://example.com/codeflow?start=17&length=9)

Do you actually delete the rule? add some comment to help reading.

Reply:
Yes. I checked the code and confirm the entry will be deleted. Also verified in test.
https://github.com/Azure/sonic-swss-common/blob/bf8c832cf1c7a6e72d7b1c843888ffb4a27088c8/common/configdb.cpp#L96